File: template_go/src/driver/sftp/s3/repository.go
package awsS3

import (
	"template/pkg/domain"
	"template/util/env"

	"github.com/aws/aws-sdk-go/service/s3"
)

type Repository struct {
	env    *env.Env
	client *s3.S3
}

func NewRepository(
	env *env.Env,
	client *s3.S3,
) *Repository {
	return &Repository{
		env:    env,
		client: client,
	}
}

func (r *Repository) Upload() ([]*domain.Entity, error) {
	return nil, nil
}
func (r *Repository) Download() ([]*domain.Entity, error) {
	return nil, nil
}

File: template_rust/src/driver/queue/sqs/repository.rs
package awsSqs

import (
	"context"
	"template/pkg/queue/domain/repository"
	"template/util/env"

	"github.com/aws/aws-sdk-go/service/sqs"
)

type Repository struct {
	env    *env.Env
	client *sqs.SQS
}

func NewRepository(
	env *env.Env,
	client *sqs.SQS,
) repository.Queuer {
	return &Repository{
		env:    env,
		client: client,
	}
}

func (r *Repository) Send(ctx context.Context) error {
	return nil
}
func (r *Repository) Receive(ctx context.Context) error {
	return nil
}
